src/elysium/rpctx.cpp

```diff
@@ -1709,25 +1710,45 @@ UniValue exodus_sendspend(const UniValue& params, bool fHelp)
     RequireExistingProperty(propertyId);
     RequireExistingDenomination(propertyId, denomination);
     RequireSaneReferenceAmount(referenceAmount);
+    RequireSigmaSpendV1Feature();
 
     // create spend
     SigmaMintId mint;
     std::vector<unsigned char> payload;
 
+    // calculate reference amount
+    CBitcoinAddress address(toAddress);
+    if (referenceAmount <= 0) {
+        CScript scriptPubKey = GetScriptForDestination(CBitcoinAddress(address).Get());
+        referenceAmount = GetDustThreshold(scriptPubKey);
+    }
+
     try {
 
         bool fPadding = chainActive.Height() >= ::Params().GetConsensus().nSigmaPaddingBlock;
 
-        auto spend = wallet->CreateSigmaSpend(propertyId, denomination, fPadding);
-        mint = spend.mint;
+        auto spend = wallet->CreateSigmaSpendV1(propertyId, denomination, fPadding);
+        auto key = wallet->GetSigmaSignatureKey(spend.mint);
+        auto pubkey = key.GetPubKey();
 
+        SigmaV1SignatureBuilder sigBuilder(address, referenceAmount, spend.proof);
+        auto signature = sigBuilder.Sign(key);
+
+        SigmaV1SignatureBuilder sigVerifier(address, referenceAmount, spend.proof);
```

Comment:
Why we don't use `sigBuilder` instead?

Reply:
I name it following the role.

Reply:
Sorry, seem like I misunderstood. Fixed!